Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -191,5 +200,24 @@ - (void)reset
     SEGLog(@"[Amplitude regnerateDeviceId];");
 }
 
+#pragma utils
+
+- (void)incrementTrait:(NSString *)trait andValue:(NSString *)value
+{
+    __block BOOL isAmountSet = false;
+
+    NSArray *increments = self.settings[@"traitsToIncrement"];
+    for (NSString *increment in increments) {
+        if ([increment isEqualToString:trait]) {
+            [self.amplitude identify:[self.identify add:trait value:value]];
+            isAmountSet = @YES;
+        }
+    }
+
+    if (!isAmountSet) {
+        [self.amplitude identify:[self.identify set:trait value:value]];
+    }
+}
+
```

Comment:
Wouldn't we have to remove incremented traits from the traits object? From Amplitude's docs: "Note: If a user property is used in multiple operations on the same Identify object, then only the first operation will be saved and the rest will be ignored. " `setUserProperties` isn't called on an Identify object, so I'd expect that traits set via this method would override any traits already incremented on the Identify object.

Reply:
Clarification from Amplitude's docs: "You may use setUserProperties shorthand to set multiple user properties at once. This method is simply a wrapper around Identify.set and identify." That said, calling `setUserProperties` creates its own instance of the Identify class in Android.

Reply:
Hmm I don't call `setUserProperties` if settings has values in `traitsToIncrement` . Yup, `setUserProperties` is just a wrapper around the logic of using `set` on AMPIdentify.